Doc 781 Manage REC credentials procedure (#1466)
RedisLabs/redislabs-docs: content/platforms/kubernetes/concepts/manage_REC_credentials.md

{{<note>}}
This procedure is only supported for operator versions 6.0.20-12 and above.
{{</note>}}

## Retrieve the current username and password 

The credentials can be used to access the cluster UI or the API. Connectivity must be configured to the cluster pods using an appropriate service (or port forwarding).

1. To inspect the random username and password created by the operator during creation, use the `kubectl get secret` command.

    ```
    $ kubectl get secret rec -o jsonpath='{.data}'
    map[password:MVUyTjd1Mm0= username:ZGVtb0ByZWRpc2xhYnMuY29t]
    ```

    The output will be the encoded password and username (the values above are examples).

1. To decode the encoded password and username, use `echo` command and the password from the previous step.

    ```
    echo MVUyTjd1Mm0= | base64 --decodexc
    ```

    This outputs the password and username in plain text. In this example, the plain text password is `12345678` and the username is `demo@redislabs.com`.

Note there are other methods to decode secrets.

## Change the Redis Enterprise cluster credentials

### Replace the REC password

1. Retrieve and take note of the current password (see above)

> 1. Access the console of a pod running a Redis Enterprise cluster.

Author: ```suggestion
1. Access the console of a [pod](https://kubernetes.io/docs/concepts/workloads/pods/) running a Redis Enterprise cluster.
```

---
Title: Manage Redis Enterprise cluster (REC) credentials

> description: 

Reviewer: ```suggestion
linkTitle: Manage cluster credentials
description: 
```

Author: I have to be careful about using cluster without specifying the type of cluster. So I think "Manage REC cluster credentials bring more clarity.

Reviewer: The existing title will force a line break, which is the main reason to use use linkTitle here.  (Title is the default for the TOC when linkTitle is absent.)

weight: 93
alwaysopen: false
categories: ["Platforms"]
aliases: /platforms/kubernetes/manage_REC_credentials/
---

> The Redis Enterprise Software on Kubernetes uses a custom resource called RedisEnterpriseCluster to create a Redis Enterprise cluster(REC). At the time of creation, it generates random credentials the operator will use to perform operations on the Redis Enterprise cluster via the APIs. The credentials are saved in a Kubernetes (K8s) secret. The secret name defaults to the name of the cluster.

Reviewer 2: ```suggestion
The Redis Enterprise Software on Kubernetes uses a custom resource called RedisEnterpriseCluster to create a Redis Enterprise cluster (REC). At the time of creation, it generates random credentials the operator will use to perform operations on the Redis Enterprise cluster via the APIs. The credentials are saved in a Kubernetes (K8s) secret. The secret name defaults to the name of the cluster.
```

{{<note>}}
This procedure is only supported for operator versions 6.0.20-12 and above.
{{</note>}}

## Retrieve the current username and password 

The credentials can be used to access the cluster UI or the API. Connectivity must be configured to the cluster pods using an appropriate service (or port forwarding).

> 1. To inspect the random  username and password created by the operator during creation, use the `kubectl get secret` command.

Reviewer: ```suggestion
1. To inspect the random username and password created by the operator during creation, use the `kubectl get secret` command.
```

    ```
    $ kubectl get secret rec -o jsonpath='{.data}'
    map[password:MVUyTjd1Mm0= username:ZGVtb0ByZWRpc2xhYnMuY29t]
    ```

    The output will be the encoded password and username (the values above are examples).

> 1. To decode the encoded password and username, use `echo` command and the password from the last step.

Reviewer: ```suggestion
1. To decode the encoded password and username, use `echo` command and the password from the previous step.
```

    ```
    echo MVUyTjd1Mm0= | base64 --decodexc
    ```

>     This will output the password and username in plain text. In this example the plain text password is 12345678 and the username is demo@redislabs.com.

Reviewer: ```suggestion
    This outputs the password and username in plain text. In this example, the plain text password is `12345678` and the username is `demo@redislabs.com`.
```

Note there are other methods to decode secrets.

## Change the Redis Enterprise cluster credentials

### Replace the REC password

1. Retrieve and take note of the current password (see above)
1. Access the console of a pod running a Redis Enterprise cluster.

    ```
    kubectl exec -it <rec-resource-name>-0 bash
    ```

1. From the pod console, add the new password for the existing user.
    ```
    REC_USER="`cat /opt/redislabs/credentials/username`"; \
    REC_PASSWORD="`cat /opt/redislabs/credentials/password`"; \
    curl -k --request POST --url https://localhost:9443/v1/users/password \

>     -u "$REC_USER:$REC_PASSWORD" --header 'Content-Type: applicat

Reviewer 2: Looks like the end of the curl command is missing.

    ```

1. From outside the node pod, update the cluster credential secret:
    1. Save the existing username to a text file .
        ```
        echo -n "<current_username>" > username 
        ```

    1. Save the new password to a text file.
        ```
        echo -n "<new_password>" > password
        ```

    1. Update the cluster credential secret.
        ```
        kubectl create secret generic <cluster_secret_name> \
        --from-file=./username --from-file=./password --dry-run \
        -o yaml | kubectl apply -f 
        ```

    It may take up to five minutes for all the components to read the new password from the updated secret. 

1. Access the console of a pod running a Redis Enterprise cluster again.

    ```
    kubectl exec -it <rec-resource-name>-0 bash
    ```

 1. From within the pod, remove the previous password to ensure only the new one applies.
    ```
    REC_USER="`cat /opt/redislabs/credentials/username`"; \
    REC_PASSWORD="`cat /opt/redislabs/credentials/password`"; \
    curl -k --request DELETE --url https://localhost:9443/v1/users/password \ 

>     -u "$REC_USER:$REC_PASSWORD" --header 'Content-Type: applic

Reviewer 2: Looks like the end of this curl command is missing too.

    ```

### Replace the REC username and password

> 1. Login to the Redis Enterprise Cluster console.

Reviewer 2: ```suggestion
1. Sign in to the Redis Enterprise Cluster console.
```

> 1. Add another admin user and choose a password.

Author: ```suggestion
1. [Add another admin user]({{<relref "rs/security/admin-console-security/user-security/#configuring-users-with-roles">}}) and choose a password.
```

> 1. [Add another admin user]({{< relref "rs/security/admin-console-security/user-security.md/#configuring-users-with-roles" >}}) and choose a password.

Author: ```suggestion
1. [Add another admin user]({{< relref "/rs/security/admin-console-security/user-security.md/#configuring-users-with-roles" >}}) and choose a password.
```

> 1. [Add another admin user]({{< relref "/rs/security/admin-console-security/user-security.md/#configuring-users-with-roles" >}}) and choose a password.

Author: ```suggestion
1. [Add another admin user]({{< relref "rs/security/admin-console-security/user-security#configuring-users-with-roles" >}}) and choose a password.
```

> 1. [Add another admin user]({{< relref "content/rs/security/admin-console-security/user-security.md/#configuring-users-with-roles" >}}) and choose a password.

Author: ```suggestion
1. [Add another admin user]({{< relref "rs/security/admin-console-security/user-security.md/#configuring-users-with-roles" >}}) and choose a password.
```

1. Set the new username in the REC spec username field.

> 1. Update the cluster credential secret:
>        1. Save the existing username to a text file .
>         ```
>         echo -n "<current_username>" > username
>         ```

Reviewer 2: ```suggestion
1. Update the cluster credential secret:

       1. Save the existing username to a text file.
        ```
        echo -n "<current_username>" > username
        ```
```
Maybe try adding a newline between the top two lines. The line break wasn't working as expected in the staged preview, and the command line text wasn't being formatted as expected either. If that doesn't work, maybe try adding \<br>\</br> to force a line break?

    1. Save the new password to a text file.
        ```
        echo -n "<new_password>" > password
        ```

    1. Update the cluster credential secret.
        ```
        kubectl create secret generic <cluster_secret_name> \
        --from-file=./username --from-file=./password --dry-run \
        -o yaml | kubectl apply -f 
        ```

> 1. It may take up to five minutes for all the components to read the new password from the updated secret.

Author: ```suggestion
It may take up to five minutes for all the components to read the new password from the updated secret.
```

---
Title: Manage Redis Enterprise cluster (REC) credentials

> description: 

Author: ```suggestion
linkTitle: Manage REC credentials
```

## Change the Redis Enterprise cluster credentials

### Replace the REC password

1. Retrieve and take note of the current password (see above)
1. Access the console of a pod running a Redis Enterprise cluster.

    ```
    kubectl exec -it <rec-resource-name>-0 bash
    ```

1. From the pod console, add the new password for the existing user.
    ```
    REC_USER="`cat /opt/redislabs/credentials/username`"; \
    REC_PASSWORD="`cat /opt/redislabs/credentials/password`"; \
    curl -k --request POST --url https://localhost:9443/v1/users/password \
    -u "$REC_USER:$REC_PASSWORD" --header 'Content-Type: applicat
    ```

1. From outside the node pod, update the cluster credential secret:
    1. Save the existing username to a text file .
        ```
        echo -n "<current_username>" > username 
        ```

    1. Save the new password to a text file.
        ```
        echo -n "<new_password>" > password
        ```

    1. Update the cluster credential secret.
        ```
        kubectl create secret generic <cluster_secret_name> \
        --from-file=./username --from-file=./password --dry-run \
        -o yaml | kubectl apply -f 
        ```

>     It may take up to five minutes for all the components to read the new password from the updated secret. 

Reviewer 3: we need to be explicit here - wait for 5 minutes before moving forward. If they don't, the account might get locked and the operator would stop working.

    ```

 1. From within the pod, remove the previous password to ensure only the new one applies.
    ```
    REC_USER="`cat /opt/redislabs/credentials/username`"; \
    REC_PASSWORD="`cat /opt/redislabs/credentials/password`"; \
    curl -k --request DELETE --url https://localhost:9443/v1/users/password \ 
    -u "$REC_USER:$REC_PASSWORD" --header 'Content-Type: applic
    ```

### Replace the REC username and password

1. Sign in to the Redis Enterprise Cluster console.
1. [Add another admin user]({{< relref "rs/security/admin-console-security/user-security#configuring-users-with-roles" >}}) and choose a password.
1. Set the new username in the REC spec username field.
1. Update the cluster credential secret:

       1. Save the existing username to a text file.
        ```
        echo -n "<current_username>" > username
        ```

    1. Save the new password to a text file.
        ```
        echo -n "<new_password>" > password
        ```

    1. Update the cluster credential secret.
        ```
        kubectl create secret generic <cluster_secret_name> \
        --from-file=./username --from-file=./password --dry-run \
        -o yaml | kubectl apply -f 
        ```

> It may take up to five minutes for all the components to read the new password from the updated secret.

Reviewer 3: same as above.
